src/cpp/core/system/PosixGroup.cpp

```diff
@@ -55,10 +55,10 @@ Error groupFrom(const boost::function<int(
    int result = 0;
    do
    {
-      buffer.reserve(buffSize);
+      buffer.resize(buffSize);
 
       // attempt the read
-      result = getGroup(value, &grp, &(buffer[0]), buffSize, &temp);
+      result = getGroup(value, &grp, safe_convert::safe_vec_addr(buffer), buffer.size(), &temp);
```

Comment:
Isn't the `safe_vec_addr` unnecessary here after the call to `buffer.resize` above?